mozci/push.py

```diff
@@ -1207,12 +1223,35 @@ def _map_failing_tasks(groups):
             # Link all the failing tasks on the given groups
             return {name: self.group_summaries[name].failing_tasks for name in groups}
 
+        real_failures_to_be_retriggered = (groups_regressions & groups_high).difference(
```

Comment:
Can you add comments explaining how we decide the groups to retrigger and backfill?
You can simply copy & paste my comments from the issue and reword them if necessary

Reply:
I was writing the comments and wondered if it would be enough to link [your comment](https://github.com/mozilla/mozci/issues/654#issuecomment-1139488070) directly in the code to avoid repeating everything?

Reply:
Sounds good!